eclipse-scout-core/src/desktop/outline/OutlineAdapter.ts

```diff
@@ -251,72 +260,115 @@ export class OutlineAdapter extends TreeAdapter {
     this.modelAdapter._linkNodeWithRowLater(page);
   }
 
-  protected static override _createTreeNodeRemote(this: Tree & { modelAdapter: TreeAdapter; _createTreeNodeOrig }, nodeModel: TreeNodeModel) {
+  protected static override _createTreeNodeRemote(this: Outline & { modelAdapter: OutlineAdapter; _createTreeNodeOrig }, pageModel: PageModel) {
     // nodeType is only used for Scout Classic pages
-    if (!this.modelAdapter || !nodeModel?.nodeType) {
-      return this._createTreeNodeOrig(nodeModel);
+    if (!this.modelAdapter || !pageModel?.nodeType) {
+      return this._createTreeNodeOrig(pageModel);
     }
 
-    nodeModel = (this.modelAdapter as OutlineAdapter)._initNodeModel(nodeModel);
-
-    if (nodeModel.nodeType === 'jsPage') {
-      if (!nodeModel.jsPageObjectType) {
-        throw new Error('jsPageObjectType not set');
+    pageModel = this.modelAdapter._initNodeModel(pageModel);
+    if (pageModel.nodeType === 'jsPage') {
+      try {
+        return this.modelAdapter._createJsPage(pageModel, this._createTreeNodeOrig.bind(this));
+      } catch (error) {
+        // Create a broken page instead of showing a fatal error so the application can still be used.
+        pageModel = {
+          ...this.modelAdapter._createJsPageModel(pageModel, null),
+          objectType: Page,
+          text: this.session.text('ui.CouldNotCreateElement'),
+          iconId: icons.EXCLAMATION_MARK_CIRCLE,
+          overviewIconId: pageModel.iconId,
+          cssClass: 'broken'
+        };
+        scout.create(ErrorHandler, {displayError: false, sendError: true}).handle(error);
       }
+    }
+
+    return this._createTreeNodeOrig(pageModel);
+  }
 
-      let jsPageModel = {
-        id: nodeModel.id,
-        parent: nodeModel.parent,
-        owner: nodeModel.owner,
-        objectType: nodeModel.jsPageObjectType
-      };
-
-      if (nodeModel.jsPageModel) {
-        // If the jsPageModel contains data objects, deserialize them if possible.
-        // Create POJOs for objects whose _type attribute cannot be resolved to a class to maintain backward compatibility.
-        let deserializedJsPageModel = dataObjects.deserialize(nodeModel.jsPageModel, null, {createPojoIfDoIsUnknown: true});
-        delete deserializedJsPageModel._type; // _type should not be written to page if present
-        jsPageModel = $.extend({}, deserializedJsPageModel, jsPageModel);
+  protected _createJsPage(pageModel: PageModel, createPage: (pageModel: PageModel) => Page): Page {
+    pageModel = this._initJsPageModel(pageModel);
+    let page = createPage(pageModel);
+    if (page.classId && page.uuid && page.classId !== page.uuid) {
+      throw new Error(`ClassId and uuid don't match for page ${page.objectType}. ClassId: ${page.classId}, Uuid: ${page.uuid}`);
+    }
+    return page;
+  }
+
+  protected _initJsPageModel(pageModel: PageModel): PageModel {
+    let pageParam = dataObjects.deserialize(pageModel.pageParam);
```

Comment:
Is PageParam deserialized 2x? Here and in _initNodeModel(). Consider remove this in favor of _initNodeModel.